src/harness/sas_testcase.py

```diff
@@ -168,20 +168,30 @@ def assertChannelsContainFrequencyRange(self, channels, frequency_range):
              frequency_range['highFrequency'])
 
   def TriggerFullActivityDumpAndWaitUntilComplete(self, server_cert, server_key):
-    request_time = datetime.utcnow().replace(microsecond=0)
+    """ this function trigger SAS UUT to generate dump files \
+    and wait for the generation of them with a timeout of 7200 seconds
+    
+    Args: the certificate and the key of the server
+    
+    return : dump message"""
+    request_time = datetime.utcnow()
     self._sas_admin.TriggerFullActivityDump()
     signal.signal(signal.SIGALRM,
                   lambda signum, frame:
                   (_ for _ in ()).throw(Exception('Full Activity Dump Check Timeout')))
     # Timeout after 2 hours if it's not completed
     signal.alarm(7200)
+    dump_message = None
     # Check generation date of full activity dump 
     while True:
-       dump_message = self._sas.GetFullActivityDump( server_cert, server_key)
-       dump_time = datetime.strptime(dump_message['generationDateTime'],
+      try:
+        dump_message = self._sas.GetFullActivityDump( server_cert, server_key)
+        dump_time = datetime.strptime(dump_message['generationDateTime'],
                                                '%Y-%m-%dT%H:%M:%SZ')
-       if request_time <= dump_time:
+        if request_time > dump_time:
```

Comment:
I think the original logic is correct; what is the reason to change it?

Reply:
yes, right it was like this in this branch but it is corrected on the master.
and the merge was not good.